src/coreclr/gc/gc.cpp

```diff
@@ -18399,7 +18550,7 @@ void gc_heap::init_background_gc ()
 #endif //DOUBLY_LINKED_FL
 
     //reset the plan allocation for each segment
-    for (heap_segment* seg = generation_allocation_segment (gen); seg != ephemeral_heap_segment;
+    for (heap_segment* seg = generation_allocation_segment (gen); seg != nullptr && seg != ephemeral_heap_segment;
```

Comment:
you don't need this change anymore - my last checkin put this code under #ifndef USE_REGIONS.

Reply:
Ok - done.